freqtrade/data/metrics.py

```diff
@@ -313,31 +317,64 @@ def calculate_sortino(
     return sortino_ratio
 
 
-def calculate_sharpe(
-    trades: pd.DataFrame, min_date: datetime, max_date: datetime, starting_balance: float
-) -> float:
+def calculate_account_value_history(
+    btdata: dict[str, DataFrame],
+    trades: DataFrame,
+    min_date: datetime,
+    max_date: datetime,
+    timeframe: str,
+    stake_currency: str,
+    start_balance: float,
+    pairlist: list[str],
+) -> Series:
+    """
+    Return a Series that contains the account value (approximately) at the open time of
+    every candle stick, in stake currency.
+    """
+    index = pd.date_range(min_date, max_date, freq=timeframe_to_resample_freq(timeframe))
+    df = pd.DataFrame(index=index)
+    df[stake_currency] = float(start_balance)
+    df[pairlist] = 0.0
+    for trade in trades.itertuples():
+        for order in trade.orders:
+            filled_at = pd.Timestamp(dt_from_ts(order["order_filled_timestamp"]))
+            stake = order["safe_price"] * order["amount"]
```

Comment:
i'm not sure that's actually correct (not for what you aim to do, anyway).

My assumption (based on the lack of description) is that this refers to the discussion on discord - which in my understanding - would create a dataframe of "timerange" lenght with interval "timeframe" - and with the account value at every step of the way.

Now let's assume a trade of 4h length (1h candles)
You can use safe_price * amount for the first candle of that trade to get the initial amount
latest on the 2nd candle, we should however use one of the available prices (close/open) - as the account value (open and closed trades) did change - so the account may be 10% higher - or 10% lower. By using safe_price - it's going to be 0% - independent of actual price movements.

Reply:
The first step is to calculate the  amount of every currencies in our account，so the same rule should apply to the quote currency. The second step - multiply every currency by its open/close price and then sum them up is where the value change take effect.

Reply:
yeah i noticed - the usage is essentially explained further below 👍